### ISSUE TITLE
NameError: uninitialized constant Bundler::GemRemoteFetcher

### ISSUE THREAD

**Issue by brauliobo** [oracle/truffleruby#1332]
While running `bundle install`
```
NameError: uninitialized constant Bundler::GemRemoteFetcher 
Did you mean?  Bundler::GemRemoteFetcher                             
An error occurred while installing concurrent-ruby (1.0.5), and Bundler cannot continue.
Make sure that `gem install concurrent-ruby -v '1.0.5'` succeeds before bundling.
                                                            
In Gemfile:                                                          
  riot_js-rails was resolved to 0.7.1, which depends on       
    rails was resolved to 5.1.4, which depends on          
      actioncable was resolved to 5.1.4, which depends on
        actionpack was resolved to 5.1.4, which depends on                      
          actionview was resolved to 5.1.4, which depends on             
            rails-dom-testing was resolved to 2.0.3, which depends on
              activesupport was resolved to 5.1.4, which depends on           
                i18n was resolved to 0.9.4, which depends on           
                  concurrent-ruby   
```

**Comment by chrisseaton** [comment]
Hi can you say which version of Bundler you're using? We only support `1.16.x` at the moment.

**Comment by brauliobo** [comment]
Yep, `1.16.1`:
```
~/P/retirement ❯❯❯ gem install bundler                                                                                                                                       master ◼
Fetching: bundler-1.16.1.gem (100%)
Successfully installed bundler-1.16.1
1 gem installed

```

**Comment by chrisseaton** [comment]
Sorry can you share the `Gemfile.lock`? You can sanitise it if it still reproduces the issue.

**Comment by eregon** [comment]
@brauliobo Can you reproduce this issue with RC2, which can be installed via RVM and rbenv?
Otherwise I think we should close this.

**Comment by brauliobo** [comment]
yes, I still can reproduce it with rc3, here it is the `Gemfile.lock`:

```
GIT
  remote: https://github.com/brauliobo/useragents-rb.git
  revision: 56b112eb7a5cd5318615da785cce6bbe039c5a5c
  specs:
    useragents (0.1.4)
      nokogiri

GEM
  remote: https://rubygems.org/
  specs:
    actioncable (5.1.4)
      actionpack (= 5.1.4)
      nio4r (~> 2.0)
      websocket-driver (~> 0.6.1)
    actionmailer (5.1.4)
      actionpack (= 5.1.4)
      actionview (= 5.1.4)
      activejob (= 5.1.4)
      mail (~> 2.5, >= 2.5.4)
      rails-dom-testing (~> 2.0)
    actionpack (5.1.4)
      actionview (= 5.1.4)
      activesupport (= 5.1.4)
      rack (~> 2.0)
      rack-test (>= 0.6.3)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.0.2)
    actionview (5.1.4)
      activesupport (= 5.1.4)
      builder (~> 3.1)
      erubi (~> 1.4)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.0.3)
    active_model_serializers (0.10.6)
      actionpack (>= 4.1, < 6)
      activemodel (>= 4.1, < 6)
      case_transform (>= 0.2)
      jsonapi-renderer (>= 0.1.1.beta1, < 0.2)
    activejob (5.1.4)
      activesupport (= 5.1.4)
      globalid (>= 0.3.6)
    activemodel (5.1.4)
      activesupport (= 5.1.4)
    activerecord (5.1.4)
      activemodel (= 5.1.4)
      activesupport (= 5.1.4)
      arel (~> 8.0)
    activesupport (5.1.4)
      concurrent-ruby (~> 1.0, >= 1.0.2)
      i18n (~> 0.7)
      minitest (~> 5.1)
      tzinfo (~> 1.1)
    arel (8.0.0)
    awesome_print (1.8.0)
    babel-source (5.8.35)
    babel-transpiler (0.7.0)
      babel-source (>= 4.0, < 6)
      execjs (~> 2.0)
    better_errors (2.4.0)
      coderay (>= 1.0.0)
      erubi (>= 1.0.0)
      rack (>= 0.9.0)
    binding_of_caller (0.8.0)
      debug_inspector (>= 0.0.1)
    builder (3.2.3)
    byebug (9.1.0)
    case_transform (0.2)
      activesupport
    chronic_duration (0.10.6)
      numerizer (~> 0.1.1)
    coderay (1.1.2)
    concurrent-ruby (1.0.5)
    concurrent-ruby (1.0.5-java)
    crass (1.0.3)
    debug_inspector (0.0.3)
    domain_name (0.5.20170404)
      unf (>= 0.0.5, < 1.0.0)
    dotenv (2.2.1)
    erubi (1.7.0)
    eventmachine (1.2.5)
    eventmachine (1.2.5-java)
    execjs (2.7.0)
    ffi (1.9.18)
    ffi (1.9.18-java)
    globalid (0.4.1)
      activesupport (>= 4.2.0)
    hashie (3.5.6)
    http-cookie (1.0.3)
      domain_name (~> 0.5)
    httpclient (2.8.3)
    i18n (0.9.4)
      concurrent-ruby (~> 1.0)
    jdbc-postgres (42.1.4)
    jquery-rails (4.3.1)
      rails-dom-testing (>= 1, < 3)
      railties (>= 4.2.0)
      thor (>= 0.14, < 2.0)
    json (2.1.0)
    json (2.1.0-java)
    jsonapi-renderer (0.1.3)
    listen (3.1.5)
      rb-fsevent (~> 0.9, >= 0.9.4)
      rb-inotify (~> 0.9, >= 0.9.7)
      ruby_dep (~> 1.2)
    lodash-rails (4.17.4)
      railties (>= 3.1)
    loofah (2.2.0)
      crass (~> 1.0.2)
      nokogiri (>= 1.5.9)
    mail (2.7.0)
      mini_mime (>= 0.1.1)
    mechanize (2.7.5)
      domain_name (~> 0.5, >= 0.5.1)
      http-cookie (~> 1.0)
      mime-types (>= 1.17.2)
      net-http-digest_auth (~> 1.1, >= 1.1.1)
      net-http-persistent (~> 2.5, >= 2.5.2)
      nokogiri (~> 1.6)
      ntlm-http (~> 0.1, >= 0.1.1)
      webrobots (>= 0.0.9, < 0.2)
    method_source (0.9.0)
    mime-types (3.1)
      mime-types-data (~> 3.2015)
    mime-types-data (3.2016.0521)
    mini_mime (1.0.0)
    mini_portile2 (2.3.0)
    minitest (5.11.3)
    momentjs-rails (2.17.1)
      railties (>= 3.1)
    moving_average (0.1.1)
    net-http-digest_auth (1.4.1)
    net-http-persistent (2.9.4)
    nio4r (2.2.0)
    nio4r (2.2.0-java)
    nokogiri (1.8.2)
      mini_portile2 (~> 2.3.0)
    nokogiri (1.8.2-java)
    ntlm-http (0.1.1)
    numerizer (0.1.1)
    pg (0.21.0)
    pry (0.11.2)
      coderay (~> 1.1.0)
      method_source (~> 0.9.0)
    pry (0.11.2-java)
      coderay (~> 1.1.0)
      method_source (~> 0.9.0)
      spoon (~> 0.0)
    pry-rails (0.3.6)
      pry (>= 0.10.4)
    puma (3.10.0)
    puma (3.10.0-java)
    rack (2.0.4)
    rack-test (0.8.2)
      rack (>= 1.0, < 3)
    rails (5.1.4)
      actioncable (= 5.1.4)
      actionmailer (= 5.1.4)
      actionpack (= 5.1.4)
      actionview (= 5.1.4)
      activejob (= 5.1.4)
      activemodel (= 5.1.4)
      activerecord (= 5.1.4)
      activesupport (= 5.1.4)
      bundler (>= 1.3.0)
      railties (= 5.1.4)
      sprockets-rails (>= 2.0.0)
    rails-dom-testing (2.0.3)
      activesupport (>= 4.2.0)
      nokogiri (>= 1.6)
    rails-html-sanitizer (1.0.3)
      loofah (~> 2.0)
    railties (5.1.4)
      actionpack (= 5.1.4)
      activesupport (= 5.1.4)
      method_source
      rake (>= 0.8.7)
      thor (>= 0.18.1, < 2.0)
    rake (12.3.0)
    rb-fsevent (0.10.2)
    rb-inotify (0.9.10)
      ffi (>= 0.5.0, < 2)
    redis (4.0.1)
    rerun (0.11.0)
      listen (~> 3.0)
    riot_js-rails (0.7.1)
      execjs (~> 2)
      rails (>= 3.0, < 6.0)
    rstat (0.1.2)
    ruby_dep (1.5.0)
    sass (3.5.3)
      sass-listen (~> 4.0.0)
    sass-listen (4.0.0)
      rb-fsevent (~> 0.9, >= 0.9.4)
      rb-inotify (~> 0.9, >= 0.9.7)
    sass-rails (5.0.6)
      railties (>= 4.0.0, < 6)
      sass (~> 3.1)
      sprockets (>= 2.8, < 4.0)
      sprockets-rails (>= 2.0, < 4.0)
      tilt (>= 1.1, < 3)
    securities (2.0.1)
      nokogiri
      rails
    sequel (4.49.0)
    sequel-rails (0.9.17)
      actionpack (>= 3.2.0)
      activemodel
      railties (>= 3.2.0)
      sequel (>= 3.28, < 5.0)
    slim (3.0.8)
      temple (>= 0.7.6, < 0.9)
      tilt (>= 1.3.3, < 2.1)
    spoon (0.0.6)
      ffi
    spring (2.0.2)
      activesupport (>= 4.2)
    spring-watcher-listen (2.0.1)
      listen (>= 2.7, < 4.0)
      spring (>= 1.2, < 3.0)
    sprockets (3.7.1)
      concurrent-ruby (~> 1.0)
      rack (> 1, < 3)
    sprockets-es6 (0.9.2)
      babel-source (>= 5.8.11)
      babel-transpiler
      sprockets (>= 3.0.0)
    sprockets-rails (2.3.3)
      actionpack (>= 3.0)
      activesupport (>= 3.0)
      sprockets (>= 2.8, < 4.0)
    table_print (1.5.6)
    temple (0.8.0)
    thor (0.20.0)
    thread_safe (0.3.6)
    thread_safe (0.3.6-java)
    tilt (2.0.8)
    timecop (0.9.1)
    tzinfo (1.2.5)
      thread_safe (~> 0.1)
    uglifier (3.2.0)
      execjs (>= 0.3.0, < 3)
    unf (0.1.4)
      unf_ext
    unf (0.1.4-java)
    unf_ext (0.0.7.4)
    wamp_client (0.0.8)
      json
      websocket-eventmachine-client
    wamp_rails (0.0.2)
      wamp_client (~> 0.0.7)
    webrobots (0.1.2)
    websocket (1.2.4)
    websocket-driver (0.6.5)
      websocket-extensions (>= 0.1.0)
    websocket-driver (0.6.5-java)
      websocket-extensions (>= 0.1.0)
    websocket-eventmachine-base (1.2.0)
      eventmachine (~> 1.0)
      websocket (~> 1.0)
      websocket-native (~> 1.0)
    websocket-eventmachine-client (1.2.0)
      websocket-eventmachine-base (~> 1.0)
    websocket-extensions (0.1.3)
    websocket-native (1.0.0)
    websocket-native (1.0.0-java)

PLATFORMS
  java
  ruby

DEPENDENCIES
  active_model_serializers
  awesome_print
  better_errors
  binding_of_caller
  byebug
  chronic_duration
  dotenv
  hashie
  httpclient
  jdbc-postgres
  jquery-rails
  lodash-rails
  mechanize
  momentjs-rails
  moving_average
  pg
  pry-rails
  puma (~> 3.0)
  rails
  redis
  rerun
  riot_js-rails
  rstat
  sass-rails (~> 5.0)
  securities
  sequel-rails
  slim
  spring
  spring-watcher-listen (~> 2.0.0)
  sprockets (~> 3.0)
  sprockets-es6
  sprockets-rails (~> 2.0)
  table_print
  timecop
  uglifier (>= 1.3.0)
  useragents!
  wamp_client
  wamp_rails

BUNDLED WITH
   1.16.1
```

**Comment by phyllisstein** [comment]
Similar failure here with RC3 and Bundler 1.16.2, on `concurrent-ruby` and a few more gems:

```
NameError: uninitialized constant Bundler::GemRemoteFetcher
Did you mean?  Bundler::GemRemoteFetcher
An error occurred while installing concurrent-ruby (1.0.5), and Bundler cannot continue.
Make sure that `gem install concurrent-ruby -v '1.0.5' --source 'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  apollo_upload_server was resolved to 2.0.0.beta.1, which depends on
    rails was resolved to 5.2.0, which depends on
      actioncable was resolved to 5.2.0, which depends on
        actionpack was resolved to 5.2.0, which depends on
          actionview was resolved to 5.2.0, which depends on
            rails-dom-testing was resolved to 2.0.3, which depends on
              activesupport was resolved to 5.2.0, which depends on
                i18n was resolved to 1.0.1, which depends on
                  concurrent-ruby


NameError: uninitialized constant Bundler::GemRemoteFetcher
An error occurred while installing thread_safe (0.3.6), and Bundler cannot continue.
Make sure that `gem install thread_safe -v '0.3.6' --source 'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  apollo_upload_server was resolved to 2.0.0.beta.1, which depends on
    rails was resolved to 5.2.0, which depends on
      actioncable was resolved to 5.2.0, which depends on
        actionpack was resolved to 5.2.0, which depends on
          actionview was resolved to 5.2.0, which depends on
            rails-dom-testing was resolved to 2.0.3, which depends on
              activesupport was resolved to 5.2.0, which depends on
                tzinfo was resolved to 1.2.5, which depends on
                  thread_safe


NameError: uninitialized constant Bundler::GemRemoteFetcher
Did you mean?  Bundler::GemRemoteFetcher
An error occurred while installing erubi (1.7.1), and Bundler cannot continue.
Make sure that `gem install erubi -v '1.7.1' --source 'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  apollo_upload_server was resolved to 2.0.0.beta.1, which depends on
    rails was resolved to 5.2.0, which depends on
      actioncable was resolved to 5.2.0, which depends on
        actionpack was resolved to 5.2.0, which depends on
          actionview was resolved to 5.2.0, which depends on
            erubi


NameError: uninitialized constant Bundler::GemRemoteFetcher
An error occurred while installing crass (1.0.4), and Bundler cannot continue.
Make sure that `gem install crass -v '1.0.4' --source 'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  apollo_upload_server was resolved to 2.0.0.beta.1, which depends on
    rails was resolved to 5.2.0, which depends on
      actioncable was resolved to 5.2.0, which depends on
        actionpack was resolved to 5.2.0, which depends on
          actionview was resolved to 5.2.0, which depends on
            rails-html-sanitizer was resolved to 1.0.4, which depends on
              loofah was resolved to 2.2.2, which depends on
                crass
```

Here's my `Gemfile.lock`, in case that's helpful:

```
GIT
  remote: https://github.com/discourse/mini_mime.git
  revision: de8065c5ff5dc251502dd233b1c9070913cad642
  specs:
    mini_mime (1.0.0)

GIT
  remote: https://github.com/petergoldstein/dalli.git
  revision: 17344b625676bd1aa45f87757e0b718a3e1ae282
  specs:
    dalli (2.7.8)

PATH
  remote: gems/sally
  specs:
    sally (0.0.1)
      multi_json
      rails (~> 5.2)

PATH
  remote: gems/shrine-neo4j
  specs:
    shrine-neo4j (0.0.1)

GEM
  remote: https://rubygems.org/
  specs:
    actioncable (5.2.0)
      actionpack (= 5.2.0)
      nio4r (~> 2.0)
      websocket-driver (>= 0.6.1)
    actionmailer (5.2.0)
      actionpack (= 5.2.0)
      actionview (= 5.2.0)
      activejob (= 5.2.0)
      mail (~> 2.5, >= 2.5.4)
      rails-dom-testing (~> 2.0)
    actionpack (5.2.0)
      actionview (= 5.2.0)
      activesupport (= 5.2.0)
      rack (~> 2.0)
      rack-test (>= 0.6.3)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.0.2)
    actionview (5.2.0)
      activesupport (= 5.2.0)
      builder (~> 3.1)
      erubi (~> 1.4)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.0.3)
    activejob (5.2.0)
      activesupport (= 5.2.0)
      globalid (>= 0.3.6)
    activemodel (5.2.0)
      activesupport (= 5.2.0)
    activerecord (5.2.0)
      activemodel (= 5.2.0)
      activesupport (= 5.2.0)
      arel (>= 9.0)
    activestorage (5.2.0)
      actionpack (= 5.2.0)
      activerecord (= 5.2.0)
      marcel (~> 0.3.1)
    activesupport (5.2.0)
      concurrent-ruby (~> 1.0, >= 1.0.2)
      i18n (>= 0.7, < 2)
      minitest (~> 5.1)
      tzinfo (~> 1.1)
    apollo_upload_server (2.0.0.beta.1)
      graphql (~> 1.7)
      rails (>= 4.2)
    arel (9.0.0)
    ast (2.4.0)
    aws-eventstream (1.0.1)
    aws-partitions (1.95.0)
    aws-sdk-core (3.22.1)
      aws-eventstream (~> 1.0)
      aws-partitions (~> 1.0)
      aws-sigv4 (~> 1.0)
      jmespath (~> 1.0)
    aws-sdk-kms (1.6.0)
      aws-sdk-core (~> 3)
      aws-sigv4 (~> 1.0)
    aws-sdk-s3 (1.16.0)
      aws-sdk-core (~> 3, >= 3.21.2)
      aws-sdk-kms (~> 1)
      aws-sigv4 (~> 1.0)
    aws-sigv4 (1.0.3)
    axiom-types (0.1.1)
      descendants_tracker (~> 0.0.4)
      ice_nine (~> 0.11.0)
      thread_safe (~> 0.3, >= 0.3.1)
    binding_of_caller (0.8.0)
      debug_inspector (>= 0.0.1)
    bootsnap (1.3.0)
      msgpack (~> 1.0)
    builder (3.2.3)
    byebug (10.0.2)
    codeclimate-engine-rb (0.4.1)
      virtus (~> 1.0)
    coderay (1.1.2)
    coercible (1.0.0)
      descendants_tracker (~> 0.0.1)
    concurrent-ruby (1.0.5)
    connection_pool (2.2.2)
    coolline (0.5.0)
      unicode_utils (~> 1.4)
    crass (1.0.4)
    debug_inspector (0.0.3)
    descendants_tracker (0.0.4)
      thread_safe (~> 0.3, >= 0.3.1)
    diff-lcs (1.3)
    down (4.5.0)
    equalizer (0.0.11)
    erubi (1.7.1)
    ethon (0.11.0)
      ffi (>= 1.3.0)
    exifr (1.3.4)
    faraday (0.15.2)
      multipart-post (>= 1.2, < 3)
    faraday_middleware (0.12.2)
      faraday (>= 0.7.4, < 1.0)
    faraday_middleware-multi_json (0.0.6)
      faraday_middleware
      multi_json
    fastimage (2.1.3)
    ffaker (2.9.0)
    ffi (1.9.25)
    formatador (0.2.5)
    fspath (3.1.0)
    fuubar (2.3.1)
      rspec-core (~> 3.0)
      ruby-progressbar (~> 1.4)
    globalid (0.4.1)
      activesupport (>= 4.2.0)
    graphql (1.8.4)
    graphql-libgraphqlparser (1.3.0)
      graphql (~> 1.2)
    guard (2.14.2)
      formatador (>= 0.2.4)
      listen (>= 2.7, < 4.0)
      lumberjack (>= 1.0.12, < 2.0)
      nenv (~> 0.1)
      notiffany (~> 0.0)
      pry (>= 0.9.12)
      shellany (~> 0.0)
      thor (>= 0.18.1)
    guard-bundler (2.1.0)
      bundler (~> 1.0)
      guard (~> 2.2)
      guard-compat (~> 1.1)
    guard-compat (1.2.1)
    guard-rails (0.8.1)
      guard (~> 2.11)
      guard-compat (~> 1.0)
    guard-rspec (4.7.3)
      guard (~> 2.1)
      guard-compat (~> 1.1)
      rspec (>= 2.99.0, < 4.0)
    hiredis (0.6.1)
    httpclient (2.8.3)
    i18n (1.0.1)
      concurrent-ruby (~> 1.0)
    ice_nine (0.11.2)
    image_optim (0.26.1)
      exifr (~> 1.2, >= 1.2.2)
      fspath (~> 3.0)
      image_size (~> 1.5)
      in_threads (~> 1.3)
      progress (~> 3.0, >= 3.0.1)
    image_optim_pack (0.5.1)
      fspath (>= 2.1, < 4)
      image_optim (~> 0.19)
    image_processing (1.4.0)
      mini_magick (~> 4.0)
      ruby-vips (>= 2.0.11, < 3)
    image_size (1.5.0)
    in_threads (1.5.0)
    jaro_winkler (1.5.1)
    jmespath (1.4.0)
    json (2.1.0)
    jwt (2.1.0)
    kwalify (0.7.2)
    listen (3.1.5)
      rb-fsevent (~> 0.9, >= 0.9.4)
      rb-inotify (~> 0.9, >= 0.9.7)
      ruby_dep (~> 1.2)
    loofah (2.2.2)
      crass (~> 1.0.2)
      nokogiri (>= 1.5.9)
    lumberjack (1.0.13)
    mail (2.7.0)
      mini_mime (>= 0.1.1)
    marcel (0.3.2)
      mimemagic (~> 0.3.2)
    method_source (0.9.0)
    mimemagic (0.3.2)
    mini_magick (4.8.0)
    mini_portile2 (2.3.0)
    minitest (5.11.3)
    msgpack (1.2.4)
    multi_json (1.13.1)
    multipart-post (2.0.0)
    nenv (0.3.0)
    neo4j (9.2.4)
      activemodel (>= 4.0)
      activesupport (>= 4.0)
      neo4j-core (>= 7.2.2, < 9.0.0)
      orm_adapter (~> 0.5.0)
    neo4j-core (8.1.4)
      activesupport (>= 4.0)
      faraday (>= 0.9.0)
      faraday_middleware (>= 0.10.0)
      faraday_middleware-multi_json
      httpclient
      json
      multi_json
      typhoeus (>= 1.1.2)
    nio4r (2.3.1)
    nokogiri (1.8.4)
      mini_portile2 (~> 2.3.0)
    notiffany (0.1.1)
      nenv (~> 0.1)
      shellany (~> 0.0)
    oj (3.6.3)
    orm_adapter (0.5.0)
    parallel (1.12.1)
    parser (2.5.1.0)
      ast (~> 2.4.0)
    powerpack (0.1.2)
    progress (3.4.0)
    pry (0.11.3)
      coderay (~> 1.1.0)
      method_source (~> 0.9.0)
    pry-coolline (0.2.5)
      coolline (~> 0.5)
    pry-rails (0.3.6)
      pry (>= 0.10.4)
    pry-stack_explorer (0.4.9.2)
      binding_of_caller (>= 0.7)
      pry (>= 0.9.11)
    pry-state (0.1.10)
      pry (>= 0.9.10, < 0.12.0)
    puma (3.11.4)
    rack (2.0.5)
    rack-cors (1.0.2)
    rack-protection (2.0.3)
      rack
    rack-test (1.0.0)
      rack (>= 1.0, < 3)
    rails (5.2.0)
      actioncable (= 5.2.0)
      actionmailer (= 5.2.0)
      actionpack (= 5.2.0)
      actionview (= 5.2.0)
      activejob (= 5.2.0)
      activemodel (= 5.2.0)
      activerecord (= 5.2.0)
      activestorage (= 5.2.0)
      activesupport (= 5.2.0)
      bundler (>= 1.3.0)
      railties (= 5.2.0)
      sprockets-rails (>= 2.0.0)
    rails-dom-testing (2.0.3)
      activesupport (>= 4.2.0)
      nokogiri (>= 1.6)
    rails-html-sanitizer (1.0.4)
      loofah (~> 2.2, >= 2.2.2)
    railties (5.2.0)
      actionpack (= 5.2.0)
      activesupport (= 5.2.0)
      method_source
      rake (>= 0.8.7)
      thor (>= 0.18.1, < 2.0)
    rainbow (3.0.0)
    rake (12.3.1)
    rb-fsevent (0.10.3)
    rb-inotify (0.9.10)
      ffi (>= 0.5.0, < 2)
    rb-readline (0.5.5)
    redis (4.0.1)
    reek (5.0.1)
      codeclimate-engine-rb (~> 0.4.0)
      kwalify (~> 0.7.0)
      parser (>= 2.5.0.0, < 2.6)
      rainbow (>= 2.0, < 4.0)
    rspec (3.7.0)
      rspec-core (~> 3.7.0)
      rspec-expectations (~> 3.7.0)
      rspec-mocks (~> 3.7.0)
    rspec-core (3.7.1)
      rspec-support (~> 3.7.0)
    rspec-expectations (3.7.0)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.7.0)
    rspec-mocks (3.7.0)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.7.0)
    rspec-rails (3.7.2)
      actionpack (>= 3.0)
      activesupport (>= 3.0)
      railties (>= 3.0)
      rspec-core (~> 3.7.0)
      rspec-expectations (~> 3.7.0)
      rspec-mocks (~> 3.7.0)
      rspec-support (~> 3.7.0)
    rspec-support (3.7.1)
    rubocop (0.58.0)
      jaro_winkler (~> 1.5.1)
      parallel (~> 1.10)
      parser (>= 2.5)
      powerpack (~> 0.1)
      rainbow (>= 2.2.2, < 4.0)
      ruby-progressbar (~> 1.7)
      unicode-display_width (~> 1.0, >= 1.0.1)
    ruby-progressbar (1.9.0)
    ruby-vips (2.0.12)
      ffi (~> 1.9)
    ruby_dep (1.5.0)
    shellany (0.0.1)
    shrine (2.11.0)
      down (~> 4.1)
    sidekiq (5.1.3)
      concurrent-ruby (~> 1.0)
      connection_pool (~> 2.2, >= 2.2.0)
      rack-protection (>= 1.5.0)
      redis (>= 3.3.5, < 5)
    spring (2.0.2)
      activesupport (>= 4.2)
    spring-commands-rspec (1.0.4)
      spring (>= 0.9.1)
    spring-watcher-listen (2.0.1)
      listen (>= 2.7, < 4.0)
      spring (>= 1.2, < 3.0)
    sprockets (3.7.2)
      concurrent-ruby (~> 1.0)
      rack (> 1, < 3)
    sprockets-rails (3.2.1)
      actionpack (>= 4.0)
      activesupport (>= 4.0)
      sprockets (>= 3.0.0)
    streamio-ffmpeg (3.0.2)
      multi_json (~> 1.8)
    thor (0.20.0)
    thread_safe (0.3.6)
    tunemygc (1.0.71)
    typhoeus (1.3.0)
      ethon (>= 0.9.0)
    tzinfo (1.2.5)
      thread_safe (~> 0.1)
    unicode-display_width (1.4.0)
    unicode_utils (1.4.0)
    virtus (1.0.5)
      axiom-types (~> 0.1)
      coercible (~> 1.0)
      descendants_tracker (~> 0.0, >= 0.0.3)
      equalizer (~> 0.0, >= 0.0.9)
    websocket-driver (0.7.0)
      websocket-extensions (>= 0.1.0)
    websocket-extensions (0.1.3)

PLATFORMS
  ruby

DEPENDENCIES
  apollo_upload_server (~> 2.0.0.alpha.6)
  aws-sdk-s3 (~> 1.16)
  bootsnap (>= 1.3.0)
  byebug
  connection_pool (~> 2.2)
  dalli!
  faraday
  fastimage (~> 2.1)
  ffaker (= 2.9)
  fuubar (~> 2.3)
  graphql (~> 1.8)
  graphql-libgraphqlparser (~> 1.3)
  guard (~> 2.14)
  guard-bundler
  guard-rails
  guard-rspec
  hiredis (~> 0.6)
  image_optim (~> 0.26)
  image_optim_pack (~> 0.5)
  image_processing (~> 1.4)
  jwt (~> 2.1)
  listen (~> 3.1)
  marcel (~> 0.3)
  mini_magick (~> 4.8)
  mini_mime!
  multi_json (~> 1.13)
  neo4j (~> 9.2)
  oj (~> 3.6)
  pry (~> 0.11)
  pry-coolline (~> 0.2)
  pry-rails (~> 0.3)
  pry-stack_explorer
  pry-state (~> 0.1)
  puma (~> 3.11)
  rack-cors
  rails (~> 5.2)
  rb-readline (~> 0.5)
  redis (~> 4.0)
  reek
  rspec (~> 3.7)
  rspec-expectations (~> 3.7)
  rspec-mocks (~> 3.7)
  rspec-rails (~> 3.7)
  rubocop
  sally!
  shrine (~> 2.11)
  shrine-neo4j!
  sidekiq (~> 5.1)
  spring
  spring-commands-rspec
  spring-watcher-listen (~> 2.0.0)
  streamio-ffmpeg (~> 3.0)
  tunemygc
  typhoeus
  tzinfo-data

BUNDLED WITH
   1.16.2
```

**Comment by phyllisstein** [comment]
Worked a lot better with `-j1`, thanks for the tip! Installation completed but for a few gems that were incompatible. Here's my `Gemfile` anyway, in case it's helpful in reproducing the error; the commented-out gems are the ones that broke on compilation.

```ruby
# frozen_string_literal: true

source 'https://rubygems.org'
git_source(:github) { |repo| "https://github.com/#{ repo }.git" }

gem 'apollo_upload_server', '~> 2.0.0.alpha.6'
gem 'aws-sdk-s3', '~> 1.16'
# gem 'bootsnap', '>= 1.3.0', require: false
gem 'connection_pool', '~> 2.2'
gem 'dalli', github: 'petergoldstein/dalli'
gem 'faraday'
gem 'fastimage', '~> 2.1'
gem 'ffaker', '2.9', require: false
gem 'graphql', '~> 1.8'
gem 'graphql-cache', '~> 0.2'
gem 'graphql-libgraphqlparser', '~> 1.3', require: 'graphql/libgraphqlparser'
# gem 'hiredis', '~> 0.6'
gem 'image_optim', '~> 0.26'
gem 'image_optim_pack', '~> 0.5'
gem 'image_processing', '~> 1.4', require: 'image_processing/vips'
gem 'jwt', '~> 2.1'
gem 'marcel', '~> 0.3'
gem 'mini_magick', '~> 4.8'
gem 'mini_mime', github: 'discourse/mini_mime'
gem 'multi_json', '~> 1.13'
gem 'neo4j', '~> 9.2'
# gem 'oj', '~> 3.6'
gem 'puma', '~> 3.11'
gem 'rack-cors'
gem 'rails', '~> 5.2'
gem 'redis', '~> 4.0', require: ['redis', 'redis/connection/hiredis']
gem 'shrine', '~> 2.11'
gem 'sidekiq', '~> 5.1'
gem 'streamio-ffmpeg', '~> 3.0'
# gem 'tunemygc'
gem 'typhoeus', require: ['typhoeus', 'typhoeus/adapters/faraday']
gem 'tzinfo-data', platforms: %i[mingw mswin x64_mingw jruby]
gem 'yajl-ruby', '~> 1.4', require: ['yajl', 'yajl/json_gem']

group :development, :test do
    gem 'byebug', platforms: %i[mri mingw x64_mingw]
    gem 'fuubar', '~> 2.3'
    gem 'guard', '~> 2.14', require: false
    gem 'guard-bundler', require: false
    gem 'guard-rails', require: false
    gem 'guard-rspec', require: false
    gem 'listen', '~> 3.1'
    gem 'pry', '~> 0.11'
    gem 'pry-coolline', '~> 0.2'
    gem 'pry-rails', '~> 0.3'
    gem 'pry-stack_explorer'
    gem 'pry-state', '~> 0.1'
    gem 'rb-readline', '~> 0.5'
    gem 'reek'
    gem 'rspec', '~> 3.7'
    gem 'rspec-expectations', '~> 3.7'
    gem 'rspec-mocks', '~> 3.7'
    gem 'rspec-rails', '~> 3.7'
    gem 'rubocop'
    gem 'spring'
    gem 'spring-commands-rspec'
    gem 'spring-watcher-listen', '~> 2.0.0'
end

gem 'sally', path: 'gems/sally'
gem 'shrine-neo4j', path: 'gems/shrine-neo4j'
```

**Comment by eregon** [comment]
I could not reproduce this, but given `Bundler::GemRemoteFetcher` is an autoload, I highly suspect this to be due to a bug in our `autoload` handling.
I fixed our `autoload`, added specs and it now passes most specs: 5145bb0f043145318958a834701213cbe90c770d.
I believe this should fix this issue, so I'll close it.
The fix will be in the next release, 1.0.0-rc5 (planned beginning of August).

**Comment by eregon** [comment]
This issue should now be fixed with TruffleRuby 1.0.0-rc5 which was just released.

**Comment by phyllisstein** [comment]
That's wonderful! Thanks so much for the followup, I appreciate it.
